tests/rptest/transactions/verifiers/stream_verifier.py

```diff
@@ -0,0 +1,322 @@
+import sys
+from rptest.util import wait_until
+
+from rptest.remote_scripts.stream_verifier_txn import COMMAND_PRODUCE, \
+    COMMAND_CONSUME, COMMAND_ATOMIC
+
+title = "StreamVerifier"
+filename = "stream_verifier_txn.py"
+available_actions = [COMMAND_PRODUCE, COMMAND_CONSUME, COMMAND_ATOMIC]
+
+
+class CrashedException(Exception):
+    pass
+
+
+class ConsistencyViolationException(Exception):
+    pass
+
+
+OUTPUT_LOG = f"/opt/remote/var/{title}.log"
+
+
+class StreamVerifier(Service):
+    logs = {f"{title}": {"path": OUTPUT_LOG, "collect_default": True}}
+
+    def __init__(self, context, redpanda):
+        super(StreamVerifier, self).__init__(context, num_nodes=1)
+        self._port = 8090
+        self._redpanda = redpanda
+        self._is_done = False
+        self._node = None
+        self._partitions = None
+
+    def is_alive(self, node):
+        """Check that webserver is running on target node
+
+        Args:
+            node (ClusterNode): node that is running the StreamVerifier
+
+        Returns:
+            bool: True is webserver is running
+        """
+        result = node.account.ssh_output(
+            f"bash /opt/remote/control/alive.sh {title}")
+        result = result.decode("utf-8")
+        pid_alive = "YES" in result
+
+        if not pid_alive:
+            return False
+
+        # Use requests to check if webserver returns any status
+        return len(self.get_service_config()) > 0
+
+    def is_action_ready(self, action):
+        if action not in available_actions:
+            raise RuntimeError(
+                f"'{action}' is unknown. "
+                f"Available actions are {', '.join(available_actions)}")
+        try:
+            r = self.get_produce_status()
+            if r['status'][action] == 'READY':
+                # Check that no active command is running
+                return True
+            else:
+                self.logger.debug(f"Requested action of '{action}' is not "
+                                  "ready. Current statuses are "
+                                  f"'{r['status']}' ")
+                return False
+            return True
+        except requests.exceptions.ConnectionError:
+            return False
+
+    def get_action_status(self, action) -> dict:
+        if action == COMMAND_PRODUCE:
+            return self.get_produce_status()
+        elif action == COMMAND_CONSUME:
+            return self.get_consume_status()
+        elif action == COMMAND_ATOMIC:
+            return self.get_atomic_status()
+        else:
+            # Should never happen
+            raise RuntimeError(f"Invalid action '{action}'")
+
+    def get_processed_count(self, action):
+        s = self.get_action_status(action)
+        # A bit ugly, but works
+        count = s.get('stats', {}).get('processed_messages', 0)
+        self.logger.debug(f"[{action}] ...{count} messages")
+        return count
+
+    def ensure_progress(self, action, delta, timeout_sec):
```

Comment:
unused?

Reply:
In these tests yes, but in case of using service for some node operations is useful. I would leave it.